### PR TITLE
Proposed extensions in order to incorporate storage of ADCP data in s…

### PR DESCRIPTION
**Pull request by SverreBerg**
…ub groups.

### CODE REVIEW

File: docs/tableMean_current_sub_group.adoc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ 3+|{attr}:calendar = "gregorian" 
+ 3+|{attr}:long_name = "TBD" 
+ 3+|{attr}:standard_name = "time" 
+ 3+|{attr}:units = "nanoseconds since 1601-01-01 00:00:00Z" 
+
+ |{var}float percent_good_limit |MA |Filtering parameter. Mean velocity value disgarded if quality below given limit.
+ 3+|{attr}:units = "%" 
+ 3+|{attr}:long_name = "Percent good limit used to filter the water current velocities" 
+ 3+|{attr}float :valid_range = 0.0, 100.0 
+
```
Comment (lberger29) [inline docs/tableMean_current_sub_group.adoc:104]: This should be defined in ADCP_sub_group  as it is used first for ADCP_sub_group before averaging?

**Review by lberger29 — CHANGES_REQUESTED** [review]
Correlation, Error_velocity and quality indicator would benefit with a more precise definition.

Netcdf files can be very large since current_velocity is given in three different coordinate systems (beam, vessel and geographical) and with and without averaging. Current velocities in beam and vessel coordinate system could be optionnal.

File: .vs/ProjectSettings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (cyrilleponcelet) [inline .vs/ProjectSettings.json:1]: You should remove all .vs related code from you commit.

File: .vs/VSWorkspaceState.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (cyrilleponcelet) [inline .vs/VSWorkspaceState.json:1]: You should remove all .vs related code from you commit.

File: docs/tableSonar.adoc [hunk 4]
```diff
@@ -16,7 +16,9 @@ e|Types | |
...
  2+|{var}byte enum transmit_t {CW = 0, LFM = 1, HFM = 2} |Type of transmit pulse. CW = continuous wave – a pulse nominally of one frequency, LFM = linear frequency modulation – a pulse which varies from transmit_frequency_start to transmit_frequency_stop in a linear manner over the nominal pulse duration (transmit_duration_nominal), HFM = hyperbolic frequency modulation – a pulse which varies from transmit_frequency_start to transmit_frequency_stop in a hyperbolic manner over the nominal pulse duration (transmit_duration_nominal).
+ 2+|{var}ping_time(*) ping_t |Variable length vector used to store ping time of pings used in averaging. Type added for ADCP.
  2+|{var}float(*) sample_t |Variable length vector used to store ragged arrays of backscatter data. Data type can be varied to suit data storage needs.
+ 2+|{var}float(*) sample_v |Variable length vector used to store ragged arrays of velocity data. Type added for ADCP.
  2+|{var}float(*) angle_t |Variable length vector used to store ragged arrays of split-aperture angles. Data type can varied to suit data storage needs.
```
[the remark this answers is not in the document]
Comment (cyrilleponcelet) [reply docs/tableSonar.adoc:22]: Type added for ADCP and used in ADCP subgroups only should be defined in ADCP subgroups

File: docs/tableBeamGroup1.adoc [hunk 5]
```diff
@@ -280,4 +280,8 @@ e|Variables | |
  3+|{attr}:units = "m" 
  3+|{attr}:coordinates = "ping_time platform_latitude platform_longitude" 
  2+|{attr}:vertical_coordinate_reference_system = "MSL depth" |The vertical datum to which distance are referred to. Possible values are 'MSL Depth' or 'LAT Depth'
+
+e|Subgroups | |
```
Comment (cyrilleponcelet) [inline docs/tableBeamGroup1.adoc:284]: This should be subgroup description, it has no long_name

Something like this

|Subgroups | |
 |{var}ADCP |O | Subgroup containing ADCP calculated current velocities data.

File: docs/tableADCP_sub_group.adoc [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+ 3+|{attr}:long_name = "Velocity motion stabilization applied (or not) "
+
+ |{var}float vertical_sample_interval(ping_time) |M |Distance between samples.
+ 3+|{attr}:units = "s" 
+ 3+|{attr}:long_name = "Interval between recorded data samples" 
+ 3+|{attr}float :valid_min = 0.0 
+
+e|Subgroups | |
+ |{var}Mean_current |O |Calculated averaged current velocities.
+  3+|{attr}:long_name = "Sonar/Beam_group1/ADCP/Mean_current"
```
Comment (cyrilleponcelet) [inline docs/tableADCP_sub_group.adoc:143]: This should be subgroup description 

Something like this

|Subgroups | |
 |{var}Mean_current |O | Subgroup containing calculated averaged current velocities data.

File: docs/crr341.adoc [hunk 7]
```diff
@@ -288,6 +288,23 @@ include::tableSonar.adoc[]
 [[sonarBeammodeTable]]
 include::tableBeamGroup1.adoc[]
 
+==== Proposed ADCP additions
+
+This group contains the ADCP beam data, single ping calculations and associated metadata.
+The netCDF4 group name is /ADCP.
```
Comment (cyrilleponcelet) [inline docs/crr341.adoc:294]: The netCDF4 group name is ADCP and is a subgroup of /Sonar/Beam_group

File: docs/crr341.adoc [hunk 8]
```diff
@@ -288,6 +288,23 @@ include::tableSonar.adoc[]
...
+
+.Description of the ADCP subgroup of the BeamGroup1 group.
+[[sonarADCPTable]]
+include::tableADCP_sub_group.adoc[]
+
+This group contains the calculated averaged current velocity data and associated metadata.
+The netCDF4 group name is /ADCP/Mean-current.
```
Comment (cyrilleponcelet) [inline docs/crr341.adoc:301]: The netCDF4 group name is Mean-current and is a subgroup of ADCP group

**Review by cyrilleponcelet — CHANGES_REQUESTED** [review]

**Comment by cyrilleponcelet** [comment]
Everythings seems fine to me, @gavinmacaulay @lberger29 @nilsolav  can we merge this pull request ?

**Pull request merged**